### PR TITLE
Add distance/ruler tool to image UI

### PR DESCRIPTION
**Pull request by dsmyth**
Add a ruler tool to the image UI using the vtk-js [distance widget](https://kitware.github.io/vtk-js/api/Interaction_Widgets_DistanceWidget.html)

### CODE REVIEW

**Comment by thewtex** [comment]
@oeway @floryst

**Comment by oeway** [comment]
Great work! This is exciting! Looking forward to have PaintWidget in your next PR 😉

File: src/UserInterface/createImageUI.js [hunk 1]
```diff
@@ -85,6 +86,19 @@ function createImageUI(
     );
     imageUIGroup.appendChild(volumeRenderingRow2);
 
+    // Put distance tools in their own row
+    const volumeRenderingRow3 = document.createElement('div');
+    volumeRenderingRow3.setAttribute('class', style.uiRow);
+    volumeRenderingRow3.className += ` ${viewerDOMId}-volumeRendering3 ${viewerDOMId}-toggle`;
+    volumeRenderingRow3.style.display = 'none';
+
```
[the remark this answers is not in the document]
Comment (dsmyth) [reply src/UserInterface/createImageUI.js]: Probably makes sense to put dX,Y,Z in an annotation instead. I'll take them out.
Our use case doesn't cover 2D, so I hadn't thought of that. I'll make the suggested change.

File: src/ViewerStore.js [hunk 2]
```diff
@@ -136,6 +136,13 @@ class ImageUIStore {
 
   @observable labelMapOpacity = 0.75;
   @observable labelMapCategoricalColor = 'glasbey';
+
+  distanceWidget = null;
+  distanceUpdateInProgress = false;
+  distanceEnabled = false;
+
+  @observable  distanceP1 = 0.0;
```
[the remark this answers is not in the document]
Comment (dsmyth) [reply src/ViewerStore.js]: I will fix it...

**Comment by dsmyth** [comment]
It looks like the failed test were due to an issue downloading node.js?

**Comment by thewtex** [comment]
@dsmyth thanks for the update!

> It looks like the failed tests were due to an issue downloading node.js?

I re-triggered the builds, let's see.

**Comment by thewtex** [comment]
Tested locally, works great! :100: 

image

**Pull request merged**